adapters/rubicon/rubicon_test.go

```diff
@@ -548,18 +461,340 @@ func TestAppendTracker(t *testing.T) {
 	}
 }
 
-func BenchmarkParseSizes(b *testing.B) {
-	sizes := []openrtb.Format{
-		{
-			W: 300,
-			H: 600,
-		},
+func TestNoContentResponse(t *testing.T) {
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNoContent)
+	}))
+	defer server.Close()
+
+	an, ctx, pbReq := CreatePrebidRequest(server, t)
+	_, err := an.Call(ctx, pbReq, pbReq.Bidders[0])
+
+	if err != nil {
+		t.Fatalf("Should not have gotten an error: %v", err)
+	}
+
+}
+
+func TestNotFoundResponse(t *testing.T) {
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+	}))
+	defer server.Close()
+
+	an, ctx, pbReq := CreatePrebidRequest(server, t)
+	_, err := an.Call(ctx, pbReq, pbReq.Bidders[0])
+
+	if err == nil {
+		t.Fatalf("Should have gotten an error: %v", err)
+	}
+
+}
+
+func TestWrongFormatResponse(t *testing.T) {
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("This is text."))
+	}))
+	defer server.Close()
+
+	an, ctx, pbReq := CreatePrebidRequest(server, t)
+	_, err := an.Call(ctx, pbReq, pbReq.Bidders[0])
+
+	if err == nil {
+		t.Fatalf("Should have gotten an error: %v", err)
+	}
+
+}
+
+func TestZeroSeatBidResponse(t *testing.T) {
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resp := openrtb.BidResponse{
+			ID:      "a-random-id",
+			BidID:   "another-random-id",
+			Cur:     "USD",
+			SeatBid: []openrtb.SeatBid{},
+		}
+		js, _ := json.Marshal(resp)
+		w.Write(js)
+	}))
+	defer server.Close()
+
+	an, ctx, pbReq := CreatePrebidRequest(server, t)
+	_, err := an.Call(ctx, pbReq, pbReq.Bidders[0])
+
+	if err != nil {
+		t.Fatalf("Should not have gotten an error: %v", err)
+	}
+
+}
+
+func TestEmptyBidResponse(t *testing.T) {
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resp := openrtb.BidResponse{
+			ID:    "a-random-id",
+			BidID: "another-random-id",
+			Cur:   "USD",
+			SeatBid: []openrtb.SeatBid{
+				{
+					Seat: "RUBICON",
+					Bid:  make([]openrtb.Bid, 0),
+				},
+			},
+		}
+		js, _ := json.Marshal(resp)
+		w.Write(js)
+	}))
+	defer server.Close()
+
+	an, ctx, pbReq := CreatePrebidRequest(server, t)
+	_, err := an.Call(ctx, pbReq, pbReq.Bidders[0])
+
+	if err != nil {
+		t.Fatalf("Should not have gotten an error: %v", err)
+	}
+
+}
+
+func TestWrongBidIdResponse(t *testing.T) {
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resp := openrtb.BidResponse{
+			ID:    "a-random-id",
+			BidID: "another-random-id",
```

Comment:
we should try to use ids with some kind of pattern, like test-bid-id, test-imp-id, etc. so the names are not totally random.

Reply:
id names are updated in following commit